src/main/java/com/odysseusinc/arachne/portal/api/v1/controller/BaseDataSourceController.java

```diff
@@ -107,13 +110,35 @@ public BaseDataSourceController(GenericConversionService conversionService,
             DS dataSource = convertDTOToDataSource(commonDataSourceDTO);
             dataSource.setId(dataSourceId);
             dataSource.setDataNode(exist.getDataNode());
+            dataSource.setPublished(true);
             dataSource = dataSourceService.update(dataSource);
             result = new JsonResult<>(NO_ERROR);
             result.setResult(convertDataSourceToDTO(dataSource));
         }
         return result;
     }
 
+    @RequestMapping(value = "/api/v1/data-sources/{id}/model-type", method = RequestMethod.PUT)
+    public JsonResult<DTO> updateModelType(
+            @PathVariable("id") Long dataSourceId,
+            @RequestBody DTO commonDataSourceDTO
+    ) throws IllegalAccessException, IOException, NoSuchFieldException, SolrServerException, ValidationException {
+
+        DS updating = dataSourceService.getByIdUnsecured(dataSourceId);
+        JsonResult<DTO> result = new JsonResult<>(NO_ERROR);
+        if (!Objects.equals(updating.getModelType(), commonDataSourceDTO.getModelType()) ||
```

Comment:
auto-detection of model is available only in Enterprise. call super from there

Reply:
I suggest this construction:

DS inputDS = convertDTOToDataSource(commonDataSourceDTO);
        if (dataSourceService.fieldsDefinedAtNodeAreChanged(updating, inputDS)) {

            updating = dataSourceService.updateFieldsDefinedAtNode(updating, inputDS);
            updating = dataSourceService.update(updating);
            result.setResult(convertDataSourceToDTO(updating));
        }